packages/web-components/src/components/link-list/link-list.scss

```diff
@@ -95,4 +102,24 @@
       }
     }
   }
+
+  .#{$dds-prefix}-ce--link-list__list--three-columns {
+    @include carbon--breakpoint('sm') {
+      ::slotted(#{$dds-prefix}-link-list-item) {
+        margin-right: -$carbon--layout-01;
+      }
+    }
+
+    @include carbon--breakpoint('md') {
+      display: grid;
+      grid-template-columns: 1fr 1fr 1fr;
+      grid-column-gap: 2rem;
+
+      ::slotted(#{$dds-prefix}-link-list-item) {
+        display: block;
+        margin-left: -$carbon--layout-01;
+        margin-right: 0;
+      }
+    }
+  }
```

Comment:
There seem many duplicate CSS rules with two columns mode. One way to solve it is:

```css
.#{$dds-prefix}-ce--link-list__list--split {
   /* The common rules */
}

.#{$dds-prefix}-ce--link-list__list--two-columns {
  /* 2-column mode-specific rules */
}

.#{$dds-prefix}-ce--link-list__list--three-columns {
  /* 3-column mode-specific rules */
}
```

```css
.#{$dds-prefix}-ce--link-list__list--split,
.#{$dds-prefix}-ce--link-list__list--three-columns {
   /* The common rules */
}
```